src/aws-manager.js

```diff
@@ -824,9 +825,18 @@ class AwsManager {
     }
 
     await Promise.all(worker.regions.map(async r => {
+      let allSGForRegion = [];
       await Promise.all(worker.instanceTypes.map(async t => {
         let launchSpec = launchSpecs[r.region][t.instanceType].launchSpec;
 
+        if (launchSpec.SecurityGroups) {
+          for (let sg of launchSpec.SecurityGroups) {
+            if (!_.includes(allSGForRegion, sg)) {
+              allSGForRegion.push(sg);
+            }
+          }
+        }
```

Comment:
I know I catch myself doing the same thing many times...

I've learn in JS that if it's not doing I/O, I should keep it simple as oppose to trying and be smart.